modules/apps/remote-app/remote-app-admin-web/src/main/java/com/liferay/remote/app/admin/web/internal/route/CustomElementPortletEntryFriendlyURLMapper.java

```diff
@@ -0,0 +1,100 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.remote.app.admin.web.internal.route;
+
+import com.liferay.petra.string.StringBundler;
+import com.liferay.petra.string.StringPool;
+import com.liferay.portal.kernel.portlet.DefaultFriendlyURLMapper;
+import com.liferay.portal.kernel.portlet.FriendlyURLMapper;
+import com.liferay.portal.kernel.portlet.LiferayPortletURL;
+import com.liferay.portal.kernel.portlet.Route;
+import com.liferay.portal.kernel.portlet.Router;
+import com.liferay.remote.app.admin.web.internal.portlet.CustomElementPortlet;
+import com.liferay.remote.app.model.CustomElementPortletEntry;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import javax.portlet.WindowState;
+
+/**
+ * @author Iván Zaera Avellón
+ */
+public class CustomElementPortletEntryFriendlyURLMapper
+	extends DefaultFriendlyURLMapper implements FriendlyURLMapper {
+
+	public CustomElementPortletEntryFriendlyURLMapper(
+		CustomElementPortletEntry customElementPortletEntry) {
+
+		_mapping = CustomElementPortlet.getPortletName(
+			customElementPortletEntry);
+
+		Router router = new RouterImpl();
```

Comment:
Does the URLs without the /normal/ stuff work?

It seems that you register several so it should work with only `http://localhost:8080/web/guest/home/-/vanilla_counter_40029/routing/path`  right?

Reply:
yes